### PR TITLE
refactor: Update the required APIs for audit log integration

### PR DESCRIPTION
**Pull request by rmoles**
This PR updates the required API list to only enable the APIs required for audit log. 
Mobeen provided a list of APIs targeted specifically for the audit log integration.
fixes https://lacework.atlassian.net/browse/ALLY-418
Signed-off-by: Ross Moles <ross.moles@lacework.net>

### CODE REVIEW

**Review by afiune — APPROVED** [review]
tenor-262976456

**Review by scottford-lw — APPROVED** [review]

**Comment by rmoles** [comment]
After testing these updates with the service account changes, I discovered that we need to add the `viewer` role to the SA at project level for the integration. @afiune @scottford-io, we should perhaps discuss whether we should as a minimum add the viewer role to the service-account or whether we are okay with adding it in the audit-log integration and the config integration.

File: README.md [hunk 1]
```diff
@@ -5,4 +5,50 @@
 [![GitHub release](https://img.shields.io/github/release/lacework/terraform-gcp-audit-log.svg)](https://github.com/lacework/terraform-gcp-audit-log/releases/)
 [![CircleCI status](https://circleci.com/gh/lacework/terraform-gcp-audit-log.svg?style=shield)](https://circleci.com/gh/lacework/terraform-gcp-audit-log)
 
-Terraform module for configuring an integration with Google Cloud Platform Organziations and Projects for Audit Logs analysis.
+Terraform module for configuring an integration with Google Cloud Platform Organizations and Projects for Audit Logs analysis.
+
+
+_**Note:** When `use_existing_service_account` is set to true, the required roles must be added manually. Depending on the integration type this may be at Project or Organization level._
```
Comment (afiune) [inline README.md:11]: I am hesitant to change the behavior of all the GCP modules (Config, Audit Log, and GCR),
if we are expecting the users to provide the service account and add the roles manually,
what is the sense of adding this option? If we need to go this route, let us just deprecate it
and have the user use the native Terraform resources for Lacework.

My $0.02 👆🏽

Comment (rmoles) [reply README.md:11]: Agreed. Kudos to @declanwilson-lw. He pointed me in the direction of using the google_iam_policy data provider to check if the role is already applied when using use_existing_service_account=true

File: variables.tf [hunk 2]
```diff
@@ -74,3 +71,15 @@ variable "wait_time" {
...
+  description = "Boolean for enabled Uniform Bucket Level Access on the audit log bucket"
+  type        = bool
+  default     = false
+}
+
+variable "lifecycle_rule_age" {
+  description = "Number of days to keep audit logs in Lacework GCS bucket before deleting.  Leave null to keep indefinitely"
+  type        = number
+  default     = null
+}
```
Comment (afiune) [inline variables.tf:85]: missing new line

File: README.md [hunk 3]
```diff
@@ -5,4 +5,54 @@
...
+|org_integration|If set to true, configure an organization level integration|bool|false|false|
+|organization_id|The organization ID, required if org_integration is set to true|string|""|false|
+|project_id|A project ID different from the default defined inside the provider|string|""|false|
+|use_existing_service_account|Set this to true to use an existing Service Account. When using an existing service account, the required roles must be added manually.|bool|false|false|
+|service_account_name|The Service Account name (required when use_existing_service_account is set to true). This can also be used to specify the new service account name when use_existing_service_account is set to false|string|""|false|
+|service_account_private_key|The private key in JSON format, base64 encoded (required when use_existing_service_account is set to true)|string|""|false|
+|existing_bucket_name|The name of an existing bucket you want to send the logs to|string|""|false|
+|bucket_force_destroy|Whether to force destroy the bucket and ignore any content.|bool|false|false|
+|lacework_integration_name|The integration name displayed in the Lacework UI.|string|TF GCR|false|
```
Comment (afiune) [inline README.md:42]:
```suggestion
|lacework_integration_name|The integration name displayed in the Lacework UI.|string|TF audit_log|false|
```

File: README.md [hunk 4]
```diff
@@ -5,4 +5,54 @@
...
+|required_apis|The APIs that should be enabled for this integration to be successful.|map(any)|See the Required APIs section|false|
+|prefix|The prefix that will be use at the beginning of every generated resource|string|lw-gcr|false|
```
Comment (afiune) [inline README.md:44]:
```suggestion
|prefix|The prefix that will be used at the beginning of every generated resource|string|lw-at|false|
```

File: main.tf [hunk 5]
```diff
@@ -49,20 +49,30 @@ resource "google_project_service" "required_apis" {
...
 
 resource "google_storage_bucket" "lacework_bucket" {
-  count         = length(var.existing_bucket_name) > 0 ? 0 : 1
-  project       = local.project_id
-  name          = "${var.prefix}-lacework-bucket-${random_id.uniq.hex}"
-  force_destroy = var.bucket_force_destroy
-  depends_on    = [google_project_service.required_apis]
+  count                       = length(var.existing_bucket_name) > 0 ? 0 : 1
+  project                     = local.project_id
+  name                        = "${var.prefix}-lacework-bucket-${random_id.uniq.hex}"
```
Comment (afiune) [inline main.tf:61]: Now that I think about this bucket, our prefix says `lw-at` which means that this
name will look like `lw-at-lacework-bucket-fsa213`. Feels redundant, should we do:
```suggestion
  name                        = "${var.prefix}-${random_id.uniq.hex}"
```

Comment (afiune) [reply main.tf:61]: Also, I think we have a length limit... 🤔

**Review by afiune — APPROVED** [review]
tenor-245401703

File: main.tf (hunk 5 above)
Comment (rmoles) [reply main.tf:61]: good point! Let's remove this :)

**Pull request merged**